### ISSUE TITLE
dag --dot: returns error

### ISSUE THREAD

**Issue by d-walkama** [iterative/dvc#8588]
running "dvc dag --dot" returns an error:

```unix
ERROR: unexpected error - Node names and attributes should not contain ":" unless they are quoted with "".                For example the string 'attribute:data1' should be written as '"attribute:data1"'.                Please refer https://github.com/pydot/pydot/issues/258
```

dvc doctor output:

```unix
DVC version: 2.10.2 (conda)
---------------------------------
Platform: Python 3.8.13 on Linux-5.4.0-1060-aws-x86_64-with-glibc2.10
Supports:
        webhdfs (fsspec = 2022.10.0),
        http (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        https (aiohttp = 3.8.3, aiohttp-retry = 2.8.3),
        s3 (s3fs = 2022.10.0, boto3 = 1.24.59),
        ssh (sshfs = 2022.6.0)
Cache types: hardlink, symlink
Cache directory: nfs4 on 10.20.53.109:/home
Caches: local
Remotes: local
Workspace directory: nfs4 on 10.20.53.109:/home
Repo: dvc, git

```
full debug output:

```unix
dvc dag -v --dot                                                                                                                                                           
2022-11-18 13:58:43,274 ERROR: unexpected error - Node names and attributes should not contain ":" unless they are quoted with "".                For example the string 'attribute:data1' should be written as '"attribute:data1
"'.                Please refer https://github.com/pydot/pydot/issues/258                                                                                                                                                        
------------------------------------------------------------                                                                                                                                                                     
Traceback (most recent call last):                                                                                                                                                                                               
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/dvc/cli/__init__.py", line 90, in main                                                                                                                             
    ret = cmd.do_run()                                                                                                                                                                                                           
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/dvc/cli/command.py", line 22, in do_run                                                                                                                            
    return self.run()                                                                                                                                                                                                            
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/dvc/commands/dag.py", line 133, in run                                                                                                                             
    ui.write(_show_dot(G))                                                                                                                                                                                                       
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/dvc/commands/dag.py", line 27, in _show_dot                                                                                                                        
    write_dot(G, dot_file)                                                                                                                                                                                                       
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/networkx/utils/decorators.py", line 845, in func                                                                                                                   
    return argmap._lazy_compile(__wrapper)(*args, **kwargs)                                                                                                                                                                      
  File "<class 'networkx.utils.decorators.argmap'> compilation 5", line 5, in argmap_write_dot_1                                                                                                                                 
    import itertools                                                                                                                                                                                                             
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/networkx/drawing/nx_pydot.py", line 51, in write_dot                                                                                                               
    P = to_pydot(G)                                                                                                                                                                                                              
  File "/home/dwalkama/miniconda3/lib/python3.8/site-packages/networkx/drawing/nx_pydot.py", line 263, in to_pydot                                                                                                               
    raise ValueError(                                                                                                                                                                                                            
ValueError: Node names and attributes should not contain ":" unless they are quoted with "".                For example the string 'attribute:data1' should be written as '"attribute:data1"'.                Please refer https:
//github.com/pydot/pydot/issues/258                                                              
```

**Comment by daavoo** [comment]
Hi @d-walkama , the issue was fixed in the dvc release [2.12.0](https://github.com/iterative/dvc/releases/tag/2.12.0) , could you try upgrading?

**Comment by d-walkama** [comment]
For some weird reason, just running "mamba update -c conda-forge dvc" did not update me past 2.10, but running "mamba install -c conda-forge dvc=2.34.2" did. Can confirm that this bug does not exist on newer versions of DVC.
